plugins/DNS/DNSResolver.py

```diff
@@ -0,0 +1,103 @@
+from Config import config
+
+import logging
+import time
+import json
+import re
+import os
+
+log = logging.getLogger('DNSPlugin')
+
+class DNSResolver:
+    loaded = False
+    cache = {}
+
+    def __init__(self, site_manager, nameservers, configure):
+        self.site_manager = site_manager
+        self.nameservers = nameservers
+        self.configure = configure
+
+    def load(self):
+        if not self.loaded:
+            self.loadModule()
+            self.loadCache()
+
+            self.resolver = dns.resolver.Resolver(configure=self.configure)
+
+            if not self.configure:
+                self.resolver.nameservers = self.nameservers
+
+            self.loaded = True
+
+    def loadModule(self):
+        import dns.resolver
+        import dnslink
+
+        if config.tor == 'always':
+            response = type('lamdbaobject', (object,), {})()
+            response.flags = dns.flags.TC
+
+            query = lambda *x, **y: response
+            dns.query.udp = query
+
+        globals()['dns'] = locals()['dns']
+        globals()['dnslink'] = locals()['dnslink']
```

Comment:
The `globals`/`locals` magic is not required here. Please consider using the following instead:

```python
def loadModule(self):
    global dns, dnslink
    import dns.resolver
    import dnslink

    if config.tor == 'always':
        ...
```

Reply:
Fixed!